modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1645,6 +1645,52 @@ CV__DNN_INLINE_NS_BEGIN
                              float confThreshold = 0.5f, float nmsThreshold = 0.0f);
      };
 
+    /** @brief This class represents high-level API for YOLO object detection networks.
+      *
+      * YoloDetection allows to set params for preprocessing input image.
+      * YoloDetection creates net from file with trained weights and config,
+      * sets preprocessing input, runs forward pass and return result detections.
+      */
+     class CV_EXPORTS_W_SIMPLE YoloObjectDetector : public DetectionModel
+     {
+     public:
+        /**
+          * @brief Create detection model from network represented in one of the supported formats.
+          * An order of @p model and @p config arguments does not matter.
+          * @param[in] model Binary file contains trained weights.
+          * @param[in] config Text file contains network configuration.
+          */
+        //  CV_WRAP YoloObjectDetector(const String& model, const String& config = "");
+
+         /**
+          * @brief Create model from onnx graph.
+          * @param[in] onnx_path path to onnx graph.
+          */
+         CV_WRAP YoloObjectDetector(const String& onnx, const String& yoloName);
+
+         CV_DEPRECATED_EXTERNAL  // avoid using in C++ code (need to fix bindings first)
+         YoloObjectDetector();
+
+         CV_WRAP void detect(InputArray frame, CV_OUT std::vector<int>& classIds,
```

Comment:
We should not have this here.
User uses `DetectionModel::detect()` from the base class instead. Inheritance is designed for that.

Reply:
Type of box rectangle is different from detection model's detect method.

Reply:
How user and bindings should work with that?
Inheritance is not just a word.
Why `DetectionModel` doesn't need that method overload?

Reply:
I have just checked the bindings. Call to detect method in python returns floating points for boxes as expected.
So, I do not understand, what is the issue? 
I have overridden the detect method since I changed box from `Rect` to `Rect2d`.
Do you want me to keep it as in the ancestor class? If yes, why?

Reply:
Is there any reason for object detection use Rect2d? All subsequent actions with detected object require conversion to Rect2i anyway (ROI subtraction, drawing).

Reply:
In that regard you are right, there is no point. I was coming from the point of view "what base model outputs, opencv should return it as well". You suggest to change it to  `Rect`?

Reply:
Anyway we apply some postprocessing so output is not an origin. So yes, integer type is enough for such a high level API and I cannot image the case when float/double precision is required.

Reply:
I overload detect in Imlp class. Without definition in the base class in fails to build. Could can check on your side